src/protocol/messages/mod.rs

```diff
@@ -187,7 +187,8 @@ fn read_compact_versioned_array<R: Read, T: ReadVersionedType<R>>(
     match len {
         0 => Ok(None),
         n => {
-            let len = (n - 1) as usize;
+            let len = usize::try_from(n - 1)
+                .map_err(|e| ReadVersionedError::ReadError(ReadError::Overflow(e)))?;
```

Comment:
```suggestion
            let len = usize::try_from(n - 1)
                .map_err(ReadError::Overflow)?;
```

Should be sufficient, see discussion above.